cmd/metrics-proxy/main.go

```diff
@@ -49,18 +55,18 @@ func startServer(timeout time.Duration, pw *k8s.PodScrapeWatcher) *http.Server {
 	}).Methods(http.MethodGet)
 
 	server := &http.Server{
-		Handler:      r,
-		Addr:         "0.0.0.0:15090",
-		WriteTimeout: timeout,
-		ReadTimeout:  timeout,
+		Handler:           r,
+		Addr:              fmt.Sprintf("0.0.0.0:%s", port),
+		WriteTimeout:      scrapeTimeout,
+		ReadHeaderTimeout: defaultReadTimeout,
```

Comment:
If `defaultReadTimeout < scrapeTimeout`, the context will close early.  @IbraAoad found some security reasons why it should be set, so can we modify things a bit here to reflect that and avoid the early stopping?
```suggestion
                // add comment explaining why we need this even though we don't read headers
		ReadHeaderTimeout: scrapeTimeout * 2,
```